### PR TITLE
Functional tests

### PR DESCRIPTION
**Pull request by brianstorti**
Here's a suggestion for how we could have an end-to-end tests for these handlers:

* Use rabbitmq http client to remove all the `sneakers_handlers` tests queues;
* Start some real `Sneakers` workers (in this case, 2 workers using the `DeadLetter` handler, one that succeeds and another that fails, both bound to the same routing key)
* Use `Bunny` to send a couple of messages to this exchange;
* Check that the messages were dead-lettered and are in the correct queues.

This is just a proof of concept, ideally we'd have workers to test timeouts, errors, etc.

What do you think? @abepetrillo @jjbohn

### CODE REVIEW

**Comment by jjbohn** [comment]
Nice. @abepetrillo were you working on something like this too?

@brianstorti do you think we need to actually spin up the actual runner? Would calling `Sneakers::Worker#run` work well enough? I think that would avoid any timing issues we might get into with `sleep`ing while still being a pretty true end to end test.

**Comment by brianstorti** [comment]
@jjbohn yeah, good point, I think that using `Sneaker::Worker#run` is enough. But I don't think we'll be able to get rid of that `sleep` after we publish a message with `Bunny`, though.

**Comment by abepetrillo** [comment]
Yea I was, taking a look at this now.

**Comment by abepetrillo** [comment]
Will try and adopt this in my PR for the retry worker. This is a great help!

**Comment by jjbohn** [comment]
No problem. I think `Worker#run` just calls subscribe on the bunny
connection once, so it won't create persistent runner or anything so sleep
shouldn't be necessary.

**Comment by brianstorti** [comment]
@jjbohn yeah, I mean for this [second sleep](https://github.com/alphasights/sneakers_handlers/pull/4/files#diff-24a329ac156f2a3e822fca4f32533245R30), after we publish a message. The first one we can get rid of.

**Comment by abepetrillo** [comment]
@brianstorti Attempted to use the `workers.run` suggestion without success, I think we need the fork and sleep as suggested.

**Comment by brianstorti** [comment]
@abepetrillo [this](https://github.com/alphasights/sneakers_handlers/pull/4/commits/4185e2904c6b21e2b3fb0f6e15a4c28f391d663b) seems to be working fine for me. Did you try something different?

File: test/support/dead_letter_worker_failure.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class DeadLetterWorkerSuccess
+  include Sneakers::Worker
+
+  from_queue "sneaker_handlers.dead_letter_success_test",
+    ack: true,
+    durable: false,
+    exchange: "sneakers_handlers",
+    exchange_type: :topic,
+    routing_key: "sneakers_handlers.dead_letter_test",
+    handler: SneakersHandlers::DeadLetter,
```
Comment (abepetrillo) [inline test/support/dead_letter_worker_failure.rb]: May as well get this renamed in this PR?

**Comment by abepetrillo** [comment]
:+1:

**Pull request merged**